### PR TITLE
Pract 24/authentication middleware

### PR DESCRIPTION
**Pull request by TwoTone222**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Testing Steps / QA Criteria

© 2022 GitHub, Inc.
Terms
Privacy
Security
Status

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -1,8 +1,13 @@
+require('dotenv').config({path:'./env'})
```
Comment (koral14) [inline src/app.js:1]: do we also need this line here? as we already had requested for dotenv in the server.js file...

File: src/app.js [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
+require("express-async-errors");
 const express = require('express');
 const app = express();
 const cors = require('cors')
 const favicon = require('express-favicon');
 const logger = require('morgan');
+const express = require("express");
+
+const authenticateUser = require('./middleware/authentication');
```
Comment (koral14) [inline src/app.js:10]: this line is not used in this file and the path doesn't take you to the right file (shows as error on my end)

File: src/app.js [hunk 3]
```diff
@@ -1,8 +1,13 @@
+require('dotenv').config({path:'./env'})
+require("express-async-errors");
 const express = require('express');
 const app = express();
 const cors = require('cors')
 const favicon = require('express-favicon');
 const logger = require('morgan');
+const express = require("express");
```
Comment (koral14) [inline src/app.js:8]: notFoundMiddleware and errorHandlerMiddleware need to be imported here before being used

**Review by koral14 — COMMENTED** [review]
I'm not sure, but I suppose you could pull from the main and merge it into this branch to be able to check if the middleware works with the User API, or we can do it later through another ticket. Also, I was not able to run: npm start until I replaced the "dev" script with "start".

File: src/app.js (hunk 1 above)
Comment (TwoTone222) [reply src/app.js:1]: I think you are right about this Olga, i will remove it..

File: src/app.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (TwoTone222) [reply src/app.js:8]: Got it !!

File: src/app.js (hunk 2 above)
Comment (TwoTone222) [reply src/app.js:10]: I made an app.use statement and am going to test it now before pushing..

File: src/app.js (hunk 3 above)
Comment (TwoTone222) [reply src/app.js:8]: Also made sure to import these

**Comment by TwoTone222** [comment]
> I'm not sure, but I suppose you could pull from the main and merge it into this branch to be able to check if the middleware works with the User API, or we can do it later through another ticket. Also, I was not able to run: npm start until I replaced the "dev" script with "start".

Im actually not sure why the app,js file was changed and no longer includes the start function to call the port but I will try and address it now...

**Review by koral14 — CHANGES_REQUESTED** [review]
Jukari, after I tested your branch, I noticed that you placed the middleware before the routes. It was preventing me from testing with Postman. After this small correction, the middleware should work fine. Well done

File: src/server.js [hunk 4]
```diff
@@ -1,17 +1,18 @@
-require('dotenv').config();
+const connectDB = require('../db/connect');
 const { PORT = 8000 } = process.env;
 const app = require("./app");
 const connectDB = require('./db/connect');
```
Comment (koral14) [inline src/server.js:4]: Please remove the unused one, as you are declaring it here twice.

File: db/connect.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    //   useFindAndModify: false,
+      useUnifiedTopology: true,
+    });
+    console.log('Connected to MongoDB');
+  } catch (error) {
+    console.error('Error connecting to MongoDB:', error);
+    throw error;  // Propagate the error so that the calling code can handle it
+  }
+};
+
```
Comment (koral14) [inline db/connect.js:19]: You should remove the db folder that I created earlier as you are using this one. Also remove the auth.js file from controllers folder that I created, as we don't need it anymore.

**Pull request closed without merging**